package.json

```diff
@@ -12,6 +12,9 @@
     "lint": "prettier **.{js,ts}",
     "test:unit": "jest",
     "prepare": "yarn clean && yarn lint -c && yarn test:unit && yarn build",
+    "preversion": "yarn",
+    "version": "git checkout -b $(node scripts/get_package_version) && git add .",
+    "postversion": "git push --set-upstream origin $(node scripts/get_package_version) && git push --tags",
```

Comment:
nit: it looks like this is going to modify git's state via `--set-upstream`

Reply:
That's sort of intentional. The NPM documentation claims that running `npm version major|minor|patch` runs these 3 scripts (if set). 
So:
0. NPM does not allow you to run the command when the git workspace is not clean.
1. Before bumping the version, it runs `preversion` (`yarn`) to build and test everything.
2. When bumping the version, `npm version` creates a commit with the commit message matching the version to be published. The script `version` changes the branch (so that it doesn't commit to the current branch, most likely `main`) and adds any potential changes (at the moment, there should be one, but in the future we may generate a changelog or something).
3. After the commit, it runs `postversion`. Here, I thought it might be good to push the changes to the version branch we just created and push the tag as well.

I based these steps on the example in the [NPM docs](https://docs.npmjs.com/cli/v8/commands/npm-version).

Please, let me know if it doesn't feel right.

Reply:
Carry on, ignore me!  I was doing an unsolicited review without any context because I can't help myself :smile:.

Reply:
Your input is very welcome! Thanks for commenting. I think that, at least, the reasoning is now documented in this PR.